apps/web/src/hooks/useSyncLatteChanges.ts

```diff
@@ -0,0 +1,98 @@
+import { useCurrentDocument } from '$/app/providers/DocumentProvider'
+import { useLatteChangeActions } from '$/hooks/latte'
+import { useNavigate } from '$/hooks/useNavigate'
+import { useEvents } from '$/lib/events'
+import { ROUTES } from '$/services/routes'
+import {
+  useCurrentCommit,
+  useCurrentProject,
+} from '@latitude-data/web-ui/providers'
+import { useCallback, useMemo, useRef } from 'react'
+import { useDocumentValue } from './useDocumentValueContext'
+import { useDevMode } from './useDevMode'
+
+export function useSyncLatteChanges() {
+  const router = useNavigate()
+  const { commit } = useCurrentCommit()
+  const { project } = useCurrentProject()
+  const { document } = useCurrentDocument()
```

Comment:
composition! this hook assumes it's instantiated in the right context where all this information is present.